src/scheduler.h

```diff
@@ -155,6 +157,9 @@ class SchedulerService : public Scheduler::Service {
   // List of pending pull calls.
   std::vector<std::pair<WorkerId, ObjRef> > pull_queue_;
   std::mutex pull_queue_lock_;
+  // List of failed workers
+  std::vector<std::string> failed_workers_;
```

Comment:
Workers don't really fail, tasks fail. So I think this should probably be a list of failed tasks (along with their error messages).